contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,176 @@
+collect data such as function timings, memory allocation and CPU usage in approximated 
+or exact details. 
+
+Performance profiling mostly involves sampling the CPU and collecting function timings 
+to identify portions in the program which takes up a significant amount of execution time. 
+These can be used to identify potential areas for code optimization.
+
+### Resources:
+1. [Profiling Overview](https://msdn.microsoft.com/en-us/library/bb384493(v=vs.110).aspx)
+
+<br>
+
+# Performance Profiling For The First Time
+
+Most profilers are similar in their functionalities, user interface, and use of 
+technical terms. You are encouraged to try this tutorial using your preferred profiling 
+tools on your own project, while making neccessary adjustments when required. <br>
+
+> This tutorial demonstrates using the profiler included in [Visual Studio 2015](https://www.visualstudio.com/downloads/).
+
+<br>
+
+## 1. Run a performance profiling session
+
+With your project opened in Visual Studio 2015, run the profiler found in the menu under 
+`Debug -> Performance Profiler`. 
+
+> You can follow this guide on [Creating and running a performance session](https://msdn.microsoft.com/en-us/library/ms182372.aspx#Anchor_0).
+
+Before you start, you will be asked to choose a profiling method. CPU sampling and 
+instrumentation are the two common types of profiling methods which may be the most 
+useful for you.
+
+- | CPU Sampling | Instrumentation
+--- | --- | ---
+**Description** | Collects *sample counts* for each function in the call stack at fixed CPU intervals | Collects *elapsed timings* for each function
+**Recommended for** | Finding potential areas for optimization for the first time | Investigating I/O bottlenecks
+**Not suitable for** | Profiling thread-blocking operations (such as I/O requests) as no information can be collected in such situations | Profiling entire large projects due to its large overhead and file size of the generated report
+
+Generally, you should use **CPU sampling** to get an overview of performance issues in your 
+application, and use **instrumentation** to profile specific portions of your code when needed.
+> Read more about the [differences](https://blogs.msdn.microsoft.com/ejarvi/2005/04/07/the-choice-between-sampling-and-instrumentation/) between 
+[CPU sampling](https://msdn.microsoft.com/en-us/library/dd264994.aspx#Anchor_0) and 
+[instrumentation](https://msdn.microsoft.com/en-us/library/dd264994.aspx#Anchor_1).
+
+If you are still unsure of which profiling method to use, choose **CPU sampling** as it is 
+suitable for most general uses.
+
+<br>
+
+## 2. View the performance report
+
+After your profiling session has ended, the profiler will analyze the data collected and 
+generate a comprehensive performance report, which you can explore on your own. You may 
+need to know the differences between the technical terms *inclusive and exclusive 
+data values*.
```

Comment:
Why does the reader need to know the difference? You may want to specify the rationale so that it becomes more than just a list of definitions.